### PR TITLE
Aws az down

### PR DESCRIPTION
**Pull request by damienomurchu**
## What

Add litmus experiment to simulate failure of an aws az.

## Why

Evaluate resiliency of application under test if an az goes down.

## How

- Add new litmus-go experiment for az-down
- create dummy ACLs with deny all traffic policies
- assign ACLs to subnets of az under test
- revert ACL changes after experiment concludes

### CODE REVIEW

**Comment by damienomurchu** [comment]
Thanks for the eyes @ksatchit, I'm having a little trouble verifying it on using okteto which I documented in #229 (before I created this PR).

**Comment by ksatchit** [comment]
Hi @damienomurchu ! once again, thanks for working on the experiment. Have shared some info [here](https://github.com/litmuschaos/litmus-go/issues/229#issuecomment-742572563) regarding the tests. At this point, okteto is more for repeated devtest. The tests w/ chaosexperiment and chaosengine is a separate step - not integrated w/ okteto as such.

**Comment by damienomurchu** [comment]
I've been running this locally with okteto and verified all but the destructive aspects of the experiment involving the reassignment of the blocking ACL's to take the targeted az instances out of circulation. 

The multi-az cluster I've been testing this against is due to expire, so I will test this on Monday most likely with a new cluster. All going well I'll tidy up the PR and take it out of draft.

Thanks for the useful info and fix on #229 !

**Comment by ksatchit** [comment]
Thanks @damienomurchu  !! Also Adding other community members specifically interested in this experiment for their perspective/reviews - cc: @kazukousen @suhrud-kumar

File: chaoslib/litmus/az-down/lib/az-down.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package lib
+
+import "github.com/litmuschaos/litmus-go/pkg/log"
+
+func AZDown() error {
```
Comment (damienomurchu) [inline chaoslib/litmus/az-down/lib/az-down.go:5]: Most of the az-down business logic is in the experiment right now, but will look to refactor and move here

File: experiments/kube-aws/az-down/engine.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            # set chaos interval (in sec) as desired
+            - name: CHAOS_INTERVAL
+              value: '10'
+
+            # set aws region where az is located
+            - name: AWS_REGION
+              value: 'eu-central-1'
+
+            # set key id for IAM user to be used
+            - name: AWS_IAM_KEY_ID
```
Comment (damienomurchu) [inline experiments/kube-aws/az-down/engine.yaml:38]: I was considering passing in the AWS creds needed as env vars to keep all setup in the experiment config, but I will probably mimic the approach of other experiments and grab the IAM creds from a secret instead

Comment (kazukousen) [reply experiments/kube-aws/az-down/engine.yaml:38]: I am working on this issue https://github.com/litmuschaos/litmus/issues/2368

This issue is resolved, we can pass the key to experiment job as follows:
```yaml
- name: AWS_ACCESS_KEY_ID
  valueFrom:
    secretKeyRef:
      name: aws-credentials
      key: AWS_ACCESS_KEY_ID
```

Comment (damienomurchu) [reply experiments/kube-aws/az-down/engine.yaml:42]: That sounds awesome - thanks for the heads-up @kazukousen !

Comment (damienomurchu) [reply experiments/kube-aws/az-down/engine.yaml]: I've removed these experiment vars for now, and will keep an eye on issue you mentioned

**Comment by damienomurchu** [comment]
Mostly verified against a multi-az cluster by running experiment locally with okteto. Will run against a fresh cluster later today by regular litmus workflow.

**Comment by damienomurchu** [comment]
Hmm, getting a panic over a nil map assignment I wasn't getting when running the experiment locally with okteto

```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/litmuschaos/chaos-runner/pkg/utils.(*ExperimentDetails).SetLabels(...)
	/home/circleci/go/src/github.com/litmuschaos/chaos-runner/pkg/utils/experimentHelpers.go:168
github.com/litmuschaos/chaos-runner/pkg/utils.(*ExperimentDetails).SetValueFromChaosExperiment(0xc420891ba8, 0xc4201c3a40, 0xc4202da100, 0xc420891a00, 0x1, 0x1)
	/home/circleci/go/src/github.com/litmuschaos/chaos-runner/pkg/utils/experimentHelpers.go:40 +0x1d5
github.com/litmuschaos/chaos-runner/pkg/utils.(*ExperimentDetails).SetValueFromChaosResources(0xc420891ba8, 0xc420891a00, 0xc4201c3a40, 0xc4202da100, 0xc4202da1e0, 0x1)
	/home/circleci/go/src/github.com/litmuschaos/chaos-runner/pkg/utils/experimentHelpers.go:206 +0x396
main.main()
	/home/circleci/go/src/github.com/litmuschaos/chaos-runner/bin/runner.go:56 +0x779
```

Any maps I'm using are initialised, and I'm using an image built from the same code I tested locally, but I seem to be missing something so any fresh eyes would be welcome :)

**Comment by damienomurchu** [comment]
@ksatchit I feel the PR is probably sufficiently advanced to consider taking it out of draft, but after today I'm mostly AFK for about a month so I will be guided by you as to whether I should take this out of draft before then.

**Comment by ksatchit** [comment]
Hi @damienomurchu , yes I guess we can move it out of draft/to actual PR!! Will take a closer look shortly (just wrapping up the 1.11.0 release)

**Comment by ksatchit** [comment]
Tagging @uditgaurav on the e2e failure on container-kill experiment.

**Comment by damienomurchu** [comment]
Closing this as we went a different direction on this instead, re-using a shell script using the aws-cli to perform the same steps as in the logic in this experiment. I will leave the PR branch up in case any of this work is useful to others. 

Thanks again for the help and feedback with this PR @ksatchit

**Pull request closed without merging**